draft-ietf-tls-esni.md

```diff
@@ -620,16 +623,24 @@ error code.
 
 ### HelloRetryRequest {#hrr}
 
+As required in {{send-ech}}, clients offering ECH MUST ensure that
+HelloRetryRequest-sensitive parameters applicable to TLS 1.3 and higher
+versions in ClientHelloInner match that in ClientHelloOuter. These parameters
+include:
```

Comment:
Suppose a server decrypts ClientHelloOuter and sees that its parameters differ from ClientHelloInner. Should it reject the connection or continue because we don't want the server to bother looking at the other ClientHello very much?

I ask because it occurred to me we might want to extend this in the future where the client doesn't care about the ClientHelloOuter and retry case. This is disastrous from a recoverability perspective, but if it's a predictive connection or has some other fallback available already, maybe that's fine. In that case, the ClientHelloOuter only needs to be the bare minimum to pass syntactically.

I think we can freely specify that later, so it doesn't really matter whether this is a SHOULD or MUST on the client, but the server behavior would constrain this.

Reply:
Yeah, I think leaving this as a MUST for now and possibly relaxing later is fine, provided we don't constrain ourselves by forcing a server-side check here. So, let's leave this as is?

Reply:
The server doesn't enforce this MUST, as far as I can tell. For context, note that #339 merely requires that the server checks ECH usage doesn't change across HRR.